src/Components/Server/src/Circuits/CircuitIdFactory.cs

```diff
@@ -27,29 +32,58 @@ public CircuitIdFactory(IDataProtectionProvider provider)
         // we don't care about the underlying payload, other than its uniqueness and the fact that we
         // authenticate encrypt it using data protection.
         // For validation, the fact that we can unprotect the payload is guarantee enough.
-        public string CreateCircuitId()
+        public CircuitId CreateCircuitId()
         {
-            var buffer = new byte[32];
+            var buffer = new byte[SecretLength];
             _generator.GetBytes(buffer);
-            var payload = _protector.Protect(buffer);
 
-            return Base64UrlTextEncoder.Encode(payload);
+            var id = new byte[IdLength];
```

Comment:
nit: stackalloc?

Reply:
can't do it sadly. data protection doesn't support span.